### ISSUE TITLE
get_as<bool>() always returns true

### ISSUE THREAD

**Issue by jhh** [skystrife/cpptoml#39]
I'm wondering if I'm using this wrong. This example program:

```cpp
#include "cpptoml.h"
#include <iostream>
#include <strstream>

int main() {
  std::strstream toml;
  toml << "Int = 64\nBool = false";
  auto parser = cpptoml::parser(toml);
  auto table = parser.parse();
  std::cout << *table << std::endl;
  bool b = *table->get_as<bool>("Bool");
  std::cout << "Bool = " << std::boolalpha << b << " # expect false"
            << std::endl;
}
```

outputs the following:

```
Bool = false
Int = 64

Bool = true # expect false
```

**Comment by skystrife** [comment]
Shoot, I bet the recent commit to allow type conversions in fetching ints is what broke this...

**Comment by skystrife** [comment]
Yep. This should be fixed as of 048ce87ed078f40511f184c9d00b9daafbf8cd5a. This was particularly nasty since the serialization still did the correct thing, and thus the unit tests still all passed (except for the two known failures).

...whoops.

**Comment by jhh** [comment]
Everything works on my end. Thanks very much for the assistance!
